src/topology/metric_space/lipschitz.lean

```diff
@@ -221,22 +165,99 @@ protected lemma pow {f : End α} {K} (h : lipschitz_with K f) :
 | 0       := lipschitz_with.id
 | (n + 1) := h.mul (pow n)
 
+end emetric
+
+section metric
+
+variables [metric_space α] [metric_space β] [metric_space γ] {K : ℝ≥0}
+
+protected lemma of_dist_le' {f : α → β} {K : ℝ} (h : ∀ x y, dist (f x) (f y) ≤ K * dist x y) :
+  lipschitz_with (nnreal.of_real K) f :=
+of_dist_le $ λ x y, le_trans (h x y) $
+  mul_le_mul_of_nonneg_right (nnreal.le_coe_of_real K) dist_nonneg
+
+protected lemma mk_one {f : α → β} (h : ∀ x y, dist (f x) (f y) ≤ dist x y) :
+  lipschitz_with 1 f :=
+of_dist_le $ by simpa only [nnreal.coe_one, one_mul] using h
+
+/-- For functions to `ℝ`, it suffices to prove `f x ≤ f y + K * dist x y`; this version
+doesn't assume `0≤K`. -/
+protected lemma of_le_mul_add' {f : α → ℝ} (K : ℝ≥0) (h : ∀x y, f x ≤ f y + K * dist x y) :
```

Comment:
I am surprised that `K` is in `ℝ≥0` in this lemma.

Reply:
Also, shouldn't the name be `of_le_add_mul'`? Same thing for the next lemma.

Reply:
* `ℝ≥0` : copy+paste error, fixed;
* `mul_add` vs `add_mul`: renamed.

Thank you!